torch/_subclasses/fake_tensor.py

```diff
@@ -1462,16 +1462,29 @@ def can_run_unsafe_fallback(func: OpOverload):
                 or func.name() in grandfathered_ops_FIXME
             )
 
+        def maybe_run_unsafe_fallback(error=None):
+            # no meta kernel registered, fallback to kernel for the device
+            if has_symbolic_sizes or not can_run_unsafe_fallback(func):
+                raise UnsupportedOperatorException(func)
+            if error is None:
+                error = UnsupportedOperatorException(func)
+            return run_fallback_kernel(self, func, args, kwargs, error)
+
+        # Optimization: If there is no Meta kernel, it takes a surprisingly long
+        # amount of time to catch the NotImplementedError, so we check it here.
+        if not torch._C._dispatch_has_computed_kernel_for_dispatch_key(
+            func.name(), "Meta"
+        ):
```

Comment:
Just to confirm - will this api work for all of the following meta registrations ? `python meta registrations, prims, decomps, and c++ meta fns`

Reply:
Yes. All of the listed meta registrations end up registering a implementation to the PyTorch dispatcher at DispatchKey::Meta. This API checks if there is an implementation in the PyTorch dispatcher that may be used for DispatchKey::Meta